hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -716,12 +787,69 @@ protected void putAllInternal(Map<?, ?> map) {
         }
     }
 
-    protected Future createPutAllOperationFuture(final String name, MapEntries entries, int partitionId) {
-        MapOperation op = operationProvider.createPutAllOperation(name, entries);
-        op.setPartitionId(partitionId);
-        final long size = entries.size();
+    private boolean getPutAllUseBatching(int mapSize) {
+        // we check if the feature is enabled and if the map size is bigger than a single batch per member
+        return (putAllBatchSize > 0 && mapSize > putAllBatchSize * getNodeEngine().getClusterService().getSize());
+    }
+
+    @SuppressWarnings("checkstyle:magicnumber")
+    private int getPutAllInitialSize(boolean useBatching, int mapSize, int partitionCount) {
+        if (mapSize == 1) {
+            return 1;
+        }
+        if (useBatching) {
+            return putAllBatchSize;
+        }
+        if (putAllInitialSizeFactor < 1) {
+            // this is an educated guess for the initial size of the entries per partition, depending on the map size
+            return (int) ceil(20f * mapSize / partitionCount / log10(mapSize));
```

Comment:
Since the value is so arbitrary I thought the formula would give it more context than a name and ripping it out of the formula.